### PR TITLE
feat: a more detailed description

### PR DESCRIPTION
**Pull request by yongchin0821**
just add funtion toDetails() & property parseDetails

toDetails() will return a object like

```js
  parseDetails: {
    full: "At 09:03:01 AM, only on Tuesday, only in 2022"
    hour: " 09"
    min: "03"
    mon: ""
    mond: ""
    sec: "01"
    wkd: ", only on Tuesday"
    year: ", only in 2022"
}
```

it can help us to get the more detailed description

hope it can be applied to more complex business requirement scenarios

### CODE REVIEW

File: src/expressionDescriptor.ts [hunk 1]
```diff
@@ -14,7 +14,16 @@ export class ExpressionDescriptor {
   expressionParts: string[];
   options: Options;
   i18n: Locale;
-
+  parseDetails: {
+    full?: string | null;
+    min?: string | null;
+    hour?: string | null;
+    sec?: string | null;
+    mond?: string | null;
```
Comment (bradymholt) [inline src/expressionDescriptor.ts:22]: Can we expand this so it's more descriptive - maybe `dayOfMonth`?

File: src/expressionDescriptor.ts [hunk 2]
```diff
@@ -14,7 +14,16 @@ export class ExpressionDescriptor {
   expressionParts: string[];
   options: Options;
   i18n: Locale;
-
+  parseDetails: {
+    full?: string | null;
+    min?: string | null;
+    hour?: string | null;
+    sec?: string | null;
```
Comment (bradymholt) [inline src/expressionDescriptor.ts:21]: Can we expand this so it's more descriptive - maybe `seconds`?

File: src/expressionDescriptor.ts [hunk 3]
```diff
@@ -14,7 +14,16 @@ export class ExpressionDescriptor {
   expressionParts: string[];
   options: Options;
   i18n: Locale;
-
+  parseDetails: {
+    full?: string | null;
+    min?: string | null;
+    hour?: string | null;
```
Comment (bradymholt) [inline src/expressionDescriptor.ts:20]: Can we expand this so it's more descriptive - maybe `hours`?

File: src/expressionDescriptor.ts [hunk 4]
```diff
@@ -14,7 +14,16 @@ export class ExpressionDescriptor {
   expressionParts: string[];
   options: Options;
   i18n: Locale;
-
+  parseDetails: {
+    full?: string | null;
+    min?: string | null;
```
Comment (bradymholt) [inline src/expressionDescriptor.ts:19]: Can we expand this so it's more descriptive - maybe `minutes`?

File: src/expressionDescriptor.ts [hunk 5]
```diff
@@ -14,7 +14,16 @@ export class ExpressionDescriptor {
...
+    hour?: string | null;
+    sec?: string | null;
+    mond?: string | null;
+    mon?: string | null;
```
Comment (bradymholt) [inline src/expressionDescriptor.ts:23]: Can we expand this so it's more descriptive - maybe `month`?

File: src/expressionDescriptor.ts [hunk 6]
```diff
@@ -14,7 +14,16 @@ export class ExpressionDescriptor {
...
+    wkd?: string | null;
```
Comment (bradymholt) [inline src/expressionDescriptor.ts:24]: Can we expand this so it's more descriptive - maybe `dayOfWeek`?

File: src/expressionDescriptor.ts [hunk 7]
```diff
@@ -14,7 +14,16 @@ export class ExpressionDescriptor {
   expressionParts: string[];
   options: Options;
   i18n: Locale;
-
+  parseDetails: {
+    full?: string | null;
```
Comment (bradymholt) [inline src/expressionDescriptor.ts:18]: Can we expand this so it's more descriptive and consistent - maybe `description`?

File: src/expressionDescriptor.ts [hunk 8]
```diff
@@ -65,10 +74,44 @@ export class ExpressionDescriptor {
...
+      locale: locale,
+    };
+
+    let descripter = new ExpressionDescriptor(expression, options);
+    descripter.parseDetails.full = descripter.getFullDescription();
+    descripter.getTimeOfDayDescription();
+    descripter.parseDetails.mond = descripter.getDayOfMonthDescription();
+    descripter.parseDetails.mon = descripter.getMonthDescription();
+    descripter.parseDetails.wkd = descripter.getDayOfWeekDescription();
+    descripter.parseDetails.year = descripter.getYearDescription();
```
Comment (bradymholt) [inline src/expressionDescriptor.ts:106]: In your example you show:

```
wkd: ", only on Tuesday"
year: ", only in 2022"
```

Can we remove proceeding comma and spacing to clean it up?

**Comment by yongchin0821** [comment]
sorry for my bad variable name.
you can rename the variable to make it more easier to understand.
I tried to add as little code as possible. thanks for the code optimization.

File: src/expressionDescriptor.ts (hunk 1 above)
Comment (yongchin0821) [reply src/expressionDescriptor.ts:22]: Yes. it is up to you

File: src/expressionDescriptor.ts (hunk 2 above)
Comment (yongchin0821) [reply src/expressionDescriptor.ts:21]: Yes. it is up to you

File: src/expressionDescriptor.ts (hunk 4 above)
Comment (yongchin0821) [reply src/expressionDescriptor.ts:19]: Yes. it is up to you

File: src/expressionDescriptor.ts (hunk 3 above)
Comment (yongchin0821) [reply src/expressionDescriptor.ts:20]: Yes. it is up to you

File: src/expressionDescriptor.ts (hunk 5 above)
Comment (yongchin0821) [reply src/expressionDescriptor.ts:23]: Yes. it is up to you

File: src/expressionDescriptor.ts (hunk 6 above)
Comment (yongchin0821) [reply src/expressionDescriptor.ts:24]: Yes. it is up to you

File: src/expressionDescriptor.ts (hunk 7 above)
Comment (yongchin0821) [reply src/expressionDescriptor.ts:18]: Yes. it is up to you

File: src/expressionDescriptor.ts (hunk 8 above)
Comment (yongchin0821) [reply src/expressionDescriptor.ts:106]: it's ok if needed

**Comment by bradymholt** [comment]
Thanks for the contribution!

**Pull request merged**